### PR TITLE
Fix: the inboundconnection limit filter should be placed in front of http co…

### LINKED ISSUE #14298: maxinboundconnections doesn't work with protocl http service
#### Overview of the Issue

When use `Maxinboundconnections` in the service defaults for an service with http protocol, consul generates wrong configuration which is rejected by envoy. However, changing to tcp makes it an valid configuration.

#### Reproduction Steps

```hcl
Kind      = "service-defaults"
Name      = "frontend"
Protocol  = "http"

MaxInboundConnections = 2048
```

```
consul config write ./service-defaults-frontend.hcl     
```

```
[2022-08-22 20:21:06.617][462331][warning][config] [source/common/config/grpc_subscription_impl.cc:126] 
gRPC config for type.googleapis.com/envoy.config.listener.v3.Listener rejected: Error adding/updating listener(s) 
public_listener:0.0.0.0:22004: Error: terminal filter named envoy.filters.network.http_connection_manager 
of type envoy.filters.network.http_connection_manager must be the last filter in a network filter chain.
```

### Consul info for both Client and Server

Main branch

### PR DESCRIPTION
**Pull request by huikang**
### Description

- The inboundconnection limit filter should be placed in front of http connection manager

The current implementation of xds server adds the connection limit filter at the end of the filter chain for all protocols, which doesn't work for http. This is because http_connection_manager must be the last filter in a network filter chain.

This PR moves the connection limit filter to the head of the chain to resolve the issue.

### Testing & Reproduction steps
Please refer to the steps in the linked issue #14298 

### Links

fix #14298 
#12373

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

File: agent/xds/listeners.go [hunk 1]
```diff
@@ -1990,6 +2016,7 @@ func makeTCPProxyFilter(filterName, cluster, statPrefix string) (*envoy_listener
 
 func makeConnectionLimitFilter(limit int) (*envoy_listener_v3.Filter, error) {
 	cfg := &envoy_connection_limit_v3.ConnectionLimit{
+		StatPrefix:     "inbound connection limit",
```
Comment (freddygv) [inline agent/xds/listeners.go:2019]: This should have underscores for consistency with other stats prefixes:
```suggestion
		StatPrefix:     "inbound_connection_limit",
```

**Review by freddygv — CHANGES_REQUESTED** [review]
Nice work Hui, thank you for taking care of this!

I left a few requests below.

File: agent/xds/listeners.go [hunk 2]
```diff
@@ -1132,6 +1132,21 @@ func (s *ResourceGenerator) makeInboundListener(cfgSnap *proxycfg.ConfigSnapshot
...
+				return nil, err
+			}
+			l.FilterChains = []*envoy_listener_v3.FilterChain{
+				{
+					Filters: []*envoy_listener_v3.Filter{
+						connectionLimitFilter,
+					},
+				},
+			}
+		}
```
[the remark this answers is not in the document]
Comment (huikang) [reply agent/xds/listeners.go]: @freddygv , thanks for pointing to the reference; which clarifies my doubt.

**Review by freddygv — APPROVED** [review]
LGTM

**Pull request merged**